libcfnet/tls_generic.c

```diff
@@ -724,7 +724,20 @@ int TLSRecv(SSL *ssl, char *buffer, int toget)
     assert(toget < CF_BUFSIZE);
     assert_SSLIsBlocking(ssl);
 
-    int received = SSL_read(ssl, buffer, toget);
+    int received,tries = 0,code=0;
+    do
+      {
+        received = SSL_read(ssl, buffer, toget);
+        if (received < 0)
+          {
+            code = TLSLogError(ssl, LOG_LEVEL_ERR, "SSL_read", received);
+            if (code == SSL_ERROR_WANT_READ)
+              {
+                sleep(1);
+              }
+          }
+      } while( received < 0 && code == SSL_ERROR_WANT_READ && (tries++ < 10) );
+
```

Comment:
Would it be good to have the number 10 configurable? Also if still failed after 10 tries THEN log an error which states how many tries failed?

Reply:
Yes, exactly.